### PR TITLE
Doxygen improvements for copygb2 and degrib2

### PR DESCRIPTION
**Pull request by edwardhartnett**
Part of #77 

Improvement of doxygen in copygb2 and degrib2.

### CODE REVIEW

File: src/copygb2/copygb2.f [hunk 1]
```diff
@@ -1,183 +1,170 @@
...
+!> if specified (-g option). The interpolation type defaults to
+!> bilinear but may be specified directly (-i option).  The copying
+!> may be limited to specific fields (-k option). It may also be
+!> limited to a specified subgrid of the output grid or to a subrange
+!> of the input fields (-B and -b, -A, and -K options). Fields can be
+!> identified as scalars or vectors (-v option), which are interpolated
+!> differently.  The invalid data in the output field can be filled
+!> with mask values or merged with a merge field (-M and -m options).
+!> The output GRIB2 message can also be appended to a file (-a option).
+!> Some defaults can be overridden in a namelist file (-N option).
```
Comment (BrianCurtis-NOAA) [inline src/copygb2/copygb2.f]: Some periods have two spaces after them. Change to 1

File: src/copygb2/copygb2.f [hunk 2]
```diff
@@ -1,183 +1,170 @@
...
+!>      The data are copied only where the given 
+!>      map field is on the correct side of the threshold.
+!>      The mapthreshold defaults to '>-1.e30'; in this case,
+!>      only the map field's bitmap will limit the domain.
+!>
+!>   -b mapindex   
+!>      Optional index file used to get the map field.
+!>
+!>   -B mapgrib    
+!>      GRIB file used to get the map field.  The map field
```
Comment (BrianCurtis-NOAA) [inline src/copygb2/copygb2.f]: OK I guess check the whole thing to make sure one space after a period.

File: src/copygb2/copygb2.f [hunk 3]
```diff
@@ -1,183 +1,170 @@
...
+!> ### Input Files
+!>   - unit   11    input grib file
+!>   - unit   14    map grib file
+!>   - unit   15    merge grib file
+!>   - unit   31    input grib index file
+!>   - unit   34    map grib index file
+!>   - unit   35    merge grib index file
+!>
+!> ### Output Files
+!>   - unit   51    output grib file
```
Comment (BrianCurtis-NOAA) [inline src/copygb2/copygb2.f]:
```
### File Units
#### Input
 * 11 - Input grib file
 * 14 - ... etc.
#### Output
 * 51 - Output grib file
```

File: src/copygb2/copygb2.f [hunk 4]
```diff
@@ -590,64 +561,44 @@ SUBROUTINE EUSAGE
...
+!> @param[in] kgdti integer (200) output grid definition template values
+!> @param[in] ip integer interpolation type
+!> @param[in] ipopt integer (20) interpolation options
+!> @param[in] jpdtn integer product definition template number search options
+!> @param[in] jpdt integer (100) product definition template values search options
+!> @param[in] nuv integer number of vector parameter ids
+!> @param[in] iuv integer (100) vector parameter ids
+!> @param[in] jpdsb integer (100) kpds search options (map)
+!> @param[in] jb integer flag for map optiion
+!> @param[in] jbk integer flag for map optiion
```
Comment (BrianCurtis-NOAA) [inline src/copygb2/copygb2.f]: option

File: src/copygb2/copygb2.f [hunk 5]
```diff
@@ -833,9 +784,55 @@ SUBROUTINE CPGB(LG1,LX1,LGB,LXB,LGM,LXM,LG2,
...
```
Comment (BrianCurtis-NOAA) [inline src/copygb2/copygb2.f]: option

File: src/copygb2/copygb2.f [hunk 6]
```diff
@@ -1555,41 +1442,26 @@ SUBROUTINE INTGRIB1(K1F,KGDS1F,K2F,KGDS2F,MRL,MRO,
...
         ENDIF
       ENDIF
-C - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
       END
-C-----------------------------------------------------------------------
+
+!> Return the length of a filled grid.
+!>
+!> Given a grid description section (in w3fi63 format), return the grid
```
Comment (BrianCurtis-NOAA) [inline src/copygb2/copygb2.f:1451]: w3fi63() ?

**Comment by Hang-Lei-NOAA** [comment]
After bacio v2.5.0, the lib name has been changed, by remove the _4 _d. But on CMakeLists.txt: line38 require bacio v2.4.0 .

File: src/copygb2/copygb2.f (hunk 1 above)
Comment (edwardhartnett) [reply src/copygb2/copygb2.f]: I have done this. It's going to take a few passes to get this documentation up to standard...

File: src/copygb2/copygb2.f (hunk 3 above)
Comment (edwardhartnett) [reply src/copygb2/copygb2.f]: OK

File: src/copygb2/copygb2.f (hunk 6 above)
Comment (edwardhartnett) [reply src/copygb2/copygb2.f:1451]: Are you suggesting parens? I don't know if w3fi63 is a function...

I have no idea what it means. ;-)

**Comment by edwardhartnett** [comment]
@Hang-Lei-NOAA I have a separate issue to update use of the bacio library: #82 

It will be handled in a separate PR.

File: src/copygb2/copygb2.f (hunk 5 above)
Comment (edwardhartnett) [reply src/copygb2/copygb2.f]: Fixed.

File: src/copygb2/copygb2.f (hunk 4 above)
Comment (edwardhartnett) [reply src/copygb2/copygb2.f]: Fixed

File: src/copygb2/copygb2.f (hunk 6 above)
Comment (BrianCurtis-NOAA) [reply src/copygb2/copygb2.f:1451]: Yeah I was curious if it was a function

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Pull request merged**